runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,250 @@
+package metricsview
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+const (
+	// DefaultSummaryLimit is the maximum number of dimensions to summarize.
+	DefaultSummaryLimit = 15
+	// DefaultSampleTimeWindow is the time window used for sampling values
+	DefaultSampleTimeWindow = 24 * time.Hour
+	// statsPerDimension is the number of statistics returned per dimension
+	statsPerDimension = 4
+	// summaryQueryLimit is the maximum number of rows to return in the summary query
+	summaryQueryLimit = 1
```

Comment:
The logic is hard-coded for these values (e.g. only reads on row, scans four values per dimension), they are not configurable. So I'd recommend removing them as top-level constants and inlining them.